install.sh

```diff
@@ -243,31 +248,15 @@ function extract_install () {
     mkdir "${CREW_DEST_DIR}"
     cd "${CREW_DEST_DIR}"
 
-    #Extract and install.
+    # Extract and install.
     echo_intra "Extracting ${1} ..."
-    case "${2}" in
-    *.zst)
-      if [[ -e /usr/bin/zstd ]]; then
-        tar -I /usr/bin/zstd -xpf ../"${2}"
-      elif ("${CREW_PREFIX}"/bin/zstd --version &> /dev/null); then
-        tar -I "${CREW_PREFIX}"/bin/zstd -xpf ../"${2}"
-      else
-        echo "Zstd is broken or nonfunctional, and some packages can not be extracted properly."
-        exit 1
-      fi
-    ;;
-    *.tpxz)
-      if "${CREW_PREFIX}"/bin/pixz -h &> /dev/null; then
-        tar -I "${CREW_PREFIX}"/bin/pixz -xpf ../"${2}"
-      else
-        echo "Pixz is broken or nonfunctional, and some packages can not be extracted properly."
-        exit 1
-      fi
-    ;;
-    *)
+    # This could be avoided if our zstd was compiled with lzma support.
```

Comment:
It turns out that if you build lzma into zstd... you now have new dependencies you need to satisfy to get the zstd binary to run. Each additional addition to the zstd binary expands the dependencies.

Example:
```ruby
require 'package'

class Zstd < Package
  description 'Zstandard - Fast real-time compression algorithm'
  homepage 'http://www.zstd.net'
  version '1.5.5-1' # Do not use @_ver here, it will break the installer.
  license 'BSD or GPL-2'
  compatibility 'x86_64'
  source_url 'https://github.com/facebook/zstd.git'
  git_hashtag 'v1.5.5'

  binary_url({
    x86_64: 'file:///usr/local/tmp/packages/zstd-1.5.5-1-chromeos-x86_64.tar.xz'
  })
  binary_sha256({
    x86_64: '8ff906353dbb8e8e02b7cadc1e0c856c9751c182560538724bbc0bd79b49f91e'
  })

  depends_on 'glibc' # R
  depends_on 'gcc_lib' # R
  depends_on 'lz4' # R
  depends_on 'xzutils' # R
  depends_on 'zlibpkg' # R

  no_zstd

  def self.build
    Dir.chdir 'build/meson' do
      system "meson setup #{CREW_MESON_OPTIONS} \
        -Dstatic_runtime=true \
        -Dbin_contrib=true \
        -Dzlib=enabled \
        -Dlzma=enabled \
        -Dlz4=enabled \
        builddir"
      system 'meson configure builddir'
      system "#{CREW_NINJA} -C builddir"
    end
  end

  def self.install
    Dir.chdir 'build/meson' do
      system "DESTDIR=#{CREW_DEST_DIR} #{CREW_NINJA} -C builddir install"
    end
    # Convert symlinks to hard links in libdir.
    Dir.chdir CREW_DEST_LIB_PREFIX do
      Dir['*'].each do |f|
        FileUtils.ln File.realpath(f), f, force: true if File.symlink?(f)
      end
    end
  end
end
```
Note the additional hard runtime dependencies, which would also have to be installed as bootstrap packages because of this:
```shell
lddtree `which zstd`
/usr/local/bin/zstd (interpreter => /lib64/ld-linux-x86-64.so.2)
    libzstd.so.1 => /usr/local/lib64/libzstd.so.1
    libz.so.1 => /usr/local/lib64/libz.so.1
    liblzma.so.5 => /usr/local/lib64/liblzma.so.5
    liblz4.so.1 => /usr/local/lib64/liblz4.so.1
    libpthread.so.0 => /usr/local/lib64/libpthread.so.0
    libc.so.6 => /usr/local/lib64/libc.so.6

```

Reply:
Thank you for looking into it! And yeah, it looks like building zstd with lzma is not worth it in terms of installer complexity. A static zstd might work, but right now our static builds are tied up with musl, which has its fair share of complexity. At any rate, it'd be a lot of work to save 2 lines of code, and it'd delay this PR even more. (And let's not forget that the whole purpose of this PR is to enable 7082 to save 10,000 lines of code)

So yeah, given that this PR is now finally ready to merge (all conversations resolved, working on all architectures, passing all CI), I think it's best to (right now) leave things as they are, and we can look into maybe a non-musl static zstd later. Thanks again for looking into it!